src/analysis/souffle/authorization_logic.dl

```diff
@@ -78,29 +78,29 @@ canSayRemoveTag(speaker, path, tag) :-
 // If the owner says a speaker can say removeTag, the speaker can say
 // removeTag.
 canSayRemoveTag(speaker, path, tag) :-
```

Comment:
I think we shouldn't need to have `saysCanSayHasTag` or `says_canSay_hasTag` in the raksha code. I think we only need the ones that are prefixed with `says` (but don't involve `canSay`). The rules involving canSay duplicate the logic that should already be working in the generated authorization logic code. It should be enough to just have a raksha-auth-logic interface file with:

```
.decl saysHasTag(speaker: Principal, path: AccessPath, tag: Tag)
saysHasTag(s, p t) :- says_hasTag(s, p, t).

.decl saysRemoveTag(speaker: Principal, path: AccessPath, tag: Tag)
saysRemoveTag(s, d, p, t) :- says_removeTag(s, p, t).
```

and so on for the ones beginning with says. To be complete, I think we need these rules for:
```
says_removeTag
says_ownsTag
says_hasTag
```

and I think we can delete the rules about 

```
says_canSay_removeTag
says_canSay_hasTag
```

Reply:
(And if this doesn't work you can push it back to me and I can try to debug it)

Reply:
These are being removed in the ownership PR #215. I am not sure if we can remove these relations before that without some tests failing.

Reply:
I think @aferr is right. I added the `says_canSay_removeTag` and `says_canSay_hasTag` because I didn't understand the echoing pattern for delegation. You might have to remove some rules that use them, but everything should work without them. That said, I'm not sure this is the right PR for this; this PR just renames the existing rules and makes a simple, understandable transformation on the code. That would be complicated by mixing in the more complex removal of the `says_canSay` relations from our checked-in dl files.